### PR TITLE
Fix inference regression, private model loading, and catch the timeout error when waiting for Vespa converge

### PR DESCRIPTION
**Pull request by wanliAlex**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug fix

* **What is the current behavior?** (You can also link to an open issue here)
1. we call empty cash after each vectorise -> encode call
2. There is a private model loading regression

* **What is the new behavior (if this is a feature change)?**
1. We fix the latency speed regression
2. We fix the private model loading regression
3. Catch the timeout error when waiting for Vespa converge

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)

* **Related Python client changes** (link commit/PR here)

* **Related documentation changes** (link commit/PR here)

* **Other information**:

* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: src/marqo/vespa/vespa_client.py [hunk 1]
```diff
@@ -182,11 +183,16 @@ def wait_for_application_convergence(self, timeout: int = 120) -> None:
...
+            try:
+                if self.get_application_has_converged():
+                    return
+                else:
+                    logger.debug('Waiting for Vespa application to converge')
+                    time.sleep(1)
+            # TODO Find out what exceptions is raised here
+            except (httpx.TimeoutException, httpcore.TimeoutException):
+                logger.error("Marqo timed out waiting for Vespa application to converge. Will retry.")
+                pass
```
Comment (papa99do) [inline src/marqo/vespa/vespa_client.py]: nit. `pass` is not needed here.

Comment (wanliAlex) [reply src/marqo/vespa/vespa_client.py]: removed

**Review by papa99do — DISMISSED** [review]
LGTM

File: src/marqo/s2_inference/clip_utils.py [hunk 2]
```diff
@@ -29,6 +28,7 @@
 from marqo.s2_inference.processing.custom_clip_utils import HFTokenizer, download_model
 from marqo.s2_inference.types import *
 from marqo.tensor_search.enums import ModelProperties, InferenceParams
+from marqo.tensor_search.models.private_models import ModelAuth
```
Comment (vicilliar) [inline src/marqo/s2_inference/clip_utils.py]: nit: This and the next line can be combined.

**Comment by vicilliar** [comment]
We should probably add the 3rd change "Catch the timeout error when waiting for Vespa converge" to PR name and merge commit message for easier traceability.

File: src/marqo/s2_inference/clip_utils.py (hunk 2 above)
Comment (wanliAlex) [reply src/marqo/s2_inference/clip_utils.py]: done

**Review by vicilliar — APPROVED** [review]

**Pull request merged**